Bratseth/lock tables

bratseth opened the pull request.

arnej27959 committed:
avoid starting all RPC connections simultaneously

Commit:
Log which tenants we move to none plan

Commit:
Join in stream statement

mpolden committed:
Schedule upgrade to stable OS version

bjorncs committed:
Reapply "Use provided apache http client bundle from container"

bratseth committed:
No functional changes

bratseth committed:
No functional changes

bratseth committed:
Split into tables

bratseth committed:
Split into tables

bratseth committed:
Serialize writes to each table

bratseth committed:
Serialize writes to each table

bratseth committed:
Continue on old data

bratseth committed:
Continue on old data

Commit:
Revert "Reapply "Use provided apache http client bundle from container" [run-systemtest]""

Commit:
Merge pull request #18371 from vespa-engine/revert-18369-revert-18367-revert-18366-bjorncs/reapply

Revert "Reapply "Use provided apache http client bundle from container" [run-systemtest]""

freva commented:
27 test failures in node-repository

freva commented on node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/QuestMetricsDb.java:
```diff
@@ -236,122 +199,76 @@ public void close() {
+
+    private void ensureNodeTableIsUpdated() {
+        try {
+            if (0 == engine.getStatus(newContext().getCairoSecurityContext(), new Path(), nodeTable.name)) {
```
This is just `nodeTable.exists()`, but that is already checked before calling this the method?

bratseth replied:
Yes, it does nothing now, but I need it each time a column is added.

freva replied:
Yeah, I get that, it's just that this `if` is redundant, could leave the `try` block empty until you need it

bratseth replied:
Good point, done

bjorncs committed:
Export package used by 3rd party AWS library

bjorncs committed:
Merge pull request #18372 from vespa-engine/bjorncs/fix

Export package used by 3rd party AWS library

bjorncs committed:
Merge pull request #18373 from vespa-engine/bjorncs/reapply

Reapply "Use provided apache http client bundle from container"

bjorncs committed:
Add feature flag for overriding TLS ciphers

bjorncs committed:
Remove leftovers from HTTP2 feature flag

bjorncs committed:
Merge pull request #18374 from vespa-engine/bjorncs/tls-cipher-feature-flag

Bjorncs/tls cipher feature flag

mpolden committed:
No upgrade budget in CD

baldersheim committed:
Add feature falg for controlling onnx dryrun verification.

jonmv committed:
Try MinimalH2AsyncClient (which is optimized for multiplexing)

bjorncs committed:
Revert "Reapply "Use provided apache http client bundle from container""

bjorncs committed:
Merge pull request #18378 from vespa-engine/revert-18373-bjorncs/reapply

Revert "Reapply "Use provided apache http client bundle from container""

mpolden committed:
Merge pull request #18375 from vespa-engine/mpolden/schedule-stable-os-version

Schedule upgrade to stable OS version

jonmv committed:
Set explicit port when -1

bjorncs committed:
Merge pull request #18377 from vespa-engine/jonmv/vespa-feed-client

Try MinimalH2AsyncClient (which is optimized for multiplexing)

baldersheim committed:
'dryrun' -> 'dry-run'

baldersheim committed:
Merge pull request #18376 from vespa-engine/balder/feature-flag-for-onnx-runtime-check-during-deploy

Add feature falg for controlling onnx dryrun verification.

Commit:
Add back method, still in use

bjorncs committed:
Merge pull request #18380 from vespa-engine/hmusum/revert-API-change

Add back method, still in use

arnej27959 committed:
make each ManagedRpcServer its own Task

freva committed:
Use max value for resource utilization

bjorncs committed:
Remove warning produced by hc5 doing reflective access on SSLEngineImpl

bjorncs committed:
Merge pull request #18382 from vespa-engine/bjorncs/remove-warning

Remove warning produced by hc5 doing reflective access on SSLEngineImpl

baldersheim committed:
Only log when config actually changes.

havardpe committed:
dry run onnx models on setup

freva committed:
Merge pull request #18381 from vespa-engine/freva/use-max

Use max value for resource utilization

geirst committed:
Add option to skip communicationmanager thread.

This is originally used when handling messages over mbus.

geirst committed:
Fix typo in flag name.

baldersheim committed:
Use equality operators.

baldersheim committed:
Merge pull request #18383 from vespa-engine/balder/only-log-when-something-changes

Only log when config actually changes.

baldersheim committed:
Merge pull request #18385 from vespa-engine/geirst/typo-in-skip-communicationmanager-thread-flag

Fix typo in skip-communicationmanager-thread flag name.

baldersheim committed:
Merge pull request #18386 from vespa-engine/havardpe/onnx-dry-run-impl

dry run onnx models on setup

baldersheim committed:
GC dual semicolon

jonmv committed:
Different log level for different errors

jonmv committed:
Support concatenated JSON objects as feed format

baldersheim committed:
Merge pull request #18384 from vespa-engine/geirst/vespa-feed-bm-skip-communicationmanager-thread

Add option to skip communicationmanager thread.

geirst committed:
Merge pull request #18387 from vespa-engine/balder/gc-dual-semicolon

GC dual semicolon

arnej27959 committed:
simplify to avoid problems on shutdown

mpolden committed:
Add feature flag for deferring host encryption

mpolden committed:
Rename thinPoolSizeInBase2Gb -> reservedDiskSpaceInBase2Gb

mpolden committed:
Merge DockerProvisioningTest into VirtualNodeProvisioningTest

mpolden committed:
Less Docker

jonmv committed:
Merge pull request #18388 from vespa-engine/jonmv/vespa-feed-client

Different log level for different errors

baldersheim committed:
Various cleanup after reading indexing threading code

arnej27959 committed:
remove code which is useless now

mpolden committed:
Use list flag

baldersheim committed:
'indexFieldInverter and indexFieldWriter'

baldersheim committed:
Merge pull request #18390 from vespa-engine/balder/initial-read-of-indexing-threading-code

Various cleanup after reading indexing threading code

hakonhall committed:
Merge pull request #18389 from vespa-engine/mpolden/defer-encryption

Add feature flag for deferring host encryption

arnej27959 committed:
Merge pull request #18291 from vespa-engine/arnej/wait-before-first-health-check

avoid starting all RPC connections simultaneously

freva committed:
Merge pull request #18391 from vespa-engine/mpolden/cleanup

Less Docker and thin pool

arnej27959 committed:
keep ownership in unique_ptr

* avoid doing release() in one place and then making a new
  unique_ptr later, hoping that will always be in sync

mpolden committed:
Check for unchanged version in one place

bjorncs committed:
Add logging of most important config parameters

bjorncs committed:
Remove debug logging

vekterli committed:
Avoid race condition regression introduced in #18179

We would previously check for the presence of pending null-bucket
`RequestBucketInfoCommand`s to determine if a pending cluster state
was present. We would also attempt to block all bucket delete operations
from starting if _any_ operation was pending towards that bucket on
a given node, including bucket info requests. The former was rewritten
to instead explicitly consider pending cluster state checks instead,
as checking null buckets no longer works when using stripes.

Unfortunately, due to a long-standing bug with message tracking of
`RequestBucketInfoCommand`s, these would _always_ be marked as pending
towards the null bucket. Since all ideal state ops would be blocked
by null-bucket info requests, this would avoid starting any ideal
state op as long as _any_ other op had an info request pending for
the target node. This had the desirable (but not explicitly coded for)
side effect of inhibiting bucket deletions from racing with half-finished
merge operations. It also had the undesirable effect of needlessly
blocking ops for completely unrelated buckets.

With these changes, we now explicitly handle bucket info requests for
single buckets in the `PendingMessageTracker`, allowing inhibition
of deletions to work as expected. Also add an explicit check for
pending info requests for all ideal state ops to mirror the old
behavior (but now per-bucket instead of globally...!).

arnej27959 committed:
Merge pull request #18392 from vespa-engine/arnej/remove-extra-code

remove code which is useless now

arnej27959 committed:
Merge pull request #18393 from vespa-engine/arnej/safer-pointers

keep ownership in unique_ptr

hakonhall committed:
Merge pull request #18394 from vespa-engine/mpolden/move-check

Check for unchanged version in one place

jonmv committed:
specify it's the max

jonmv committed:
Merge pull request #18395 from vespa-engine/bjorncs/vespa-hadoop

Bjorncs/vespa hadoop

mpolden committed:
Remove unnecessary NodeVersions class

jonmv committed:
Merge pull request #18396 from vespa-engine/mpolden/cleanup

Remove unnecessary NodeVersions class

bjorncs committed:
Use system default trust/key managers if no credentials provided

Previously it initialized custom managers that trusted no-one if no CA certificates where provided.

jonmv committed:
Merge pull request #18397 from vespa-engine/bjorncs/vespa-hadoop

Use system default trust/key managers if no credentials provided

arnej27959 committed:
we must now be prepared for multiple notifyOk calls

* ManagedRpcServer now schedules its own health checks,
  so we must handle repeated OK notifications.

arnej27959 committed:
Merge pull request #18398 from vespa-engine/arnej/fix-coredump-in-doPending

we must now be prepared for multiple notifyOk calls

geirst committed:
Prepare baseline utils and tests for a single distributor stripe.

This is copied from DistributorTestUtil and LegacyDistributorTest,
and adjusted to work with one distributor stripe.

jonmv committed:
More CLI print options

bjorncs committed:
Implement dryrun in FeedClient, use it from CLI and Hadoop feeder

bjorncs committed:
Add temporary logging

jonmv committed:
Address reeview :)

jonmv committed:
Merge pull request #18401 from vespa-engine/bjorncs/vespa-hadoop

Implement dryrun in FeedClient, use it from CLI and Hadoop feeder

jonmv committed:
Update abi spec and expected test output

jonmv committed:
All exceptional failures are now exceptions

jonmv committed:
Update abi spec

Commit:
Only log if we expire something, return 1 for success

Commit:
Emable move-search-definitions-to-schemas-dir bydefault

Commit:
Merge pull request #18314 from vespa-engine/ogronnesby/trial-expirer-logging

Log which tenants we expire

jonmv committed:
Lower log level in HttpRequestStrategy

jonmv committed:
Error response is not neceesarily a _parse_ exception

bjorncs committed:
Merge pull request #18400 from vespa-engine/jonmv/vespa-feed-client

More CLI print options

Commit:
Merge pull request #18403 from vespa-engine/hmusum/emable-move-search-definitions-to-schemas-dir-by-default

Emable move-search-definitions-to-schemas-dir by default

toregge committed:
Expose proton document db heart beat age metric.

Commit:
Don't swallow exception when unable to find file references in use

Also keep 10 unused file references at all times

Commit:
Fail if unable to get active application

Commit:
Merge pull request #18404 from vespa-engine/hmusum/dont-ignore-exception-when-finding-file-references-in-use

Don't swallow exception when unable to find file references in use

vekterli committed:
Merge pull request #18402 from vespa-engine/vekterli/fix-racing-maintenance-scheduling-regression

Avoid race condition regression introduced in #18179 [run-systemtest].

bjorncs committed:
Fix invalid single document parsing

jonmv committed:
Merge pull request #18405 from vespa-engine/bjorncs/vespa-feed-client

Fix invalid single document parsing

baldersheim committed:
Merge pull request #18406 from vespa-engine/toregge/expose-heart-beat-age-metric

Expose proton document db heart beat age metric.

Commit:
Always move searchdefinitions to schemas dir

Do it by default, stop using feature flag

jonmv committed:
Propagate last retried result when aborting execution (and fix shutdown)

baldersheim committed:
Merge pull request #18407 from vespa-engine/hmusum/always-move-search-definitions-to-schemas-dir

Always move searchdefinitions to schemas dir

jonmv committed:
Avoid race with NPE outcome eduring concurrent completion

jonmv committed:
Merge pull request #18408 from vespa-engine/jonmv/vespa-feed-client

Jonmv/vespa feed client

mpolden committed:
Remove obsolete version check

bjorncs committed:
Revert "Add temporary logging"

This reverts commit a21f9bb32a568e26c46a4af8e953f218206de797.

jonmv committed:
Merge pull request #18411 from vespa-engine/bjorncs/remove-debug-setup

Revert "Add temporary logging"

jonmv committed:
Merge pull request #18410 from vespa-engine/mpolden/remove-obsolete-version-check

Remove obsolete version check

jonmv committed:
Move stuff around, change some doc, default benchmark on

bjorncs committed:
Ignore irrelevant parameters in new feed client record writer

jonmv committed:
Add interface that was moved up

jonmv committed:
Merge pull request #18413 from vespa-engine/bjorncs/remove-debug-setup

Ignore irrelevant parameters in new feed client record writer

jonmv committed:
Update more ABI spec

jonmv committed:
Merge pull request #18412 from vespa-engine/jonmv/vespa-feed-client

Move stuff around, change some doc, default benchmark on

geirst committed:
Merge pull request #18409 from vespa-engine/geirst/baseline-utils-and-tests-for-distributor-stripe

Prepare baseline utils and tests for a single distributor stripe.

bratseth committed:
Remove redundant check

bratseth committed:
Remove redundant check

bratseth committed:
Merge branch 'bratseth/lock-tables' of github.com:vespa-engine/vespa into bratseth/lock-tables
